src/rgw/rgw_sync.cc

```diff
@@ -1576,10 +1571,6 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
                                                &max_marker, INCREMENTAL_MAX_ENTRIES,
                                                &log_entries, &truncated));
           for (log_iter = log_entries.begin(); log_iter != log_entries.end(); ++log_iter) {
-            if (!period_marker.empty() && period_marker < log_iter->id) {
```

Comment:
it looks like my diagnosis in this commit was wrong - this line is not comparing markers from different periods. rather, if the old and new master zones agree on the last marker of this period, we won't see any log entries with a marker greater than `period_marker`, and this block will never execute